Frameworks/include/CGImageInternal.h

```diff
@@ -96,31 +104,262 @@ typedef enum {
+        return properties->bitsPerComponent;
+    }
+
+    inline CGBitmapInfo BitmapInfo() const {
+        const __CGImagePixelProperties* properties = Properties();
+        RETURN_RESULT_IF_NULL(properties, 0);
+        return properties->bitmapInfo;
+    }
+
+    inline CGImageAlphaInfo AlphaInfo() const {
+        return static_cast<CGImageAlphaInfo>(BitmapInfo() & kCGBitmapAlphaInfoMask);
+    }
+
+    inline CGColorSpaceRef ColorSpace() const {
+        if(_impl.colorSpace) {
+            return _impl.colorSpace;
+        }
+        
+        const __CGImagePixelProperties* properties = Properties();
+        RETURN_NULL_IF(!properties);
+        return properties->colorSpaceModel;        
+    }
+};
+
+struct __CGImage : CoreFoundation::CppBase<__CGImage, __CGImageImpl> {
+    inline Microsoft::WRL::ComPtr<IWICBitmapSource>& ImageSource() {
+        return _impl.bitmapImageSource;
+    }
+
+    inline size_t Height() const {
+        return _impl.Height();
+    }
+
+    inline size_t Width() const {
+        return _impl.Width();
+    }
+
+    inline bool IsMask() const {
+        return _impl.isMask;
+    }
+
+    inline bool Interpolate() const {
+        return _impl.interpolate;
+    }
+
+    inline CGColorSpaceRef ColorSpace() const {
+        return _impl.ColorSpace();
+    }
+
+    inline CGColorRenderingIntent RenderingIntent() const {
+        return _impl.renderingIntent;
+    }
+
+    inline CGBitmapInfo BitmapInfo() const {
+        return _impl.BitmapInfo();
+    }
+
+    inline CGImageAlphaInfo AlphaInfo() const {
+        return _impl.AlphaInfo();
+    }
+
+    inline size_t BitsPerPixel() const {
+        return _impl.BitsPerPixel();
+    }
+
+    inline size_t BitsPerComponent() const {
+        return _impl.BitsPerComponent();
+    }
+
+    inline __CGImage& SetIsMask(bool mask) {
+        _impl.isMask = mask;
+        return *this;
+    }
+
+    inline __CGImage& SetInterpolate(bool interpolate) {
+        _impl.interpolate = interpolate;
+        return *this;
+    }
+
+    inline __CGImage& SetColorSpace(CGColorSpaceRef space) {
+        if (_impl.colorSpace) {
+            CGColorSpaceRelease(_impl.colorSpace);
+        }
+        _impl.colorSpace = space;
+        CGColorSpaceRetain(space);
+        return *this;
+    }
+
+    inline __CGImage& SetRenderingIntent(CGColorRenderingIntent intent) {
+        _impl.renderingIntent = intent;
+        return *this;
+    }
+
+    //---Old to be removed //
+    CGImageBacking* _img;
     bool _has32BitAlpha;
     CGImageType _imgType;
     idretain _provider;
 
-    __CGImage();
-    ~__CGImage();
-
     inline CGImageBacking* Backing() const {
         return _img;
     }
-    CGImageBacking* DetachBacking(CGImageRef newParent);
+
+    inline CGImageBacking* DetachBacking(CGImageRef newParent) {
+        CGImageBacking* ret = _img;
+
+        _img->_parent = newParent;
+        _img = NULL;
+
+        return ret;
+    }
+    //--End Old code //
```

Comment:
not as of now, there are still some code, that's using it. Given this is WIP. yes eventually., 

---
In reply to: [84763290](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84763290) [](ancestors = 84763290)